src/locations/views.py

```diff
@@ -57,6 +62,80 @@ def get_queryset(self):
             ).filter(search=str(search_phrase))
         return queryset
 
+    @action(methods=['post'], detail=True, permission_classes=[AllowAny])
+    def flag(self, request, pk=None):
+        """
+        Flag a location
+        """
+        location = self.get_object()
+
+        admin_location_url, community_public_url, manager_location_url, tpl_data = location_data(location)
+
+        tpl_data["reason"] = request.data.get("text")
+        tpl_data["email"] = request.data.get("email")
+        tpl_data["phone"] = request.data.get("phone")
+
+        # mail the community managers
+        email = AnymailMessage(
+            template_id="d-0027bd8f47014d0f8db249acd97939ce",
+            subject="A fost raportată o locație",
+            body=f"Tocmai a fost raportată locația {location.name} din comunitatea ta <a href='{community_public_url}'>{location.community.name}</a>. \n"
+                 f"Te rog să verifici și să o publici dacă este corectă. \n"
+                 f"Poți face acest lucru <a href='{manager_location_url}'>aici</a>",
+            from_email=settings.DEFAULT_FROM_EMAIL,
+            to=location.community.admin_users_emails(),
+            bcc=[settings.SUPERADMIN_CONTACT_EMAIL],
+        )
+        email.template_id = "d-0027bd8f47014d0f8db249acd97939ce"
+        email.dynamic_template_data = tpl_data
+        email.merge_global_data = tpl_data
+        try_send_email(email)
+
+        # mail admin
+        phone = f" (tel: {request.data.get('phone')})" if request.data.get("phone") else ""
+        email = AnymailMessage(
+            template_id="d-a7a22635a4744c829b951760d76ac8f4",
+            subject="flag locatie",
+            body=f"Locația '{location.name}' din comunitatea {location.community.name} "
+                 f"a fost raportată de {request.data.get('email')}{phone} pentru următorul motiv: \n\n"
+                 f"<b>{request.data.get('text')}</b>\n\n"
```

Comment:
I would propose sanitizing the data we get here, to prevent XSS attacks. We could just use character escaping like this with the standard library without adding further dependencies:

```python
# ---snip---

import html

# ---snip---
                 f"<b>{html.escape(request.data.get('text', ''))}</b>\n\n"
# ---snip---
```

What do you think?